sharding/types.go

```diff
@@ -23,23 +23,14 @@ package sharding
 // HashFn is a sharding hash function
 type HashFn func(identifer string) uint32
 
-// ShardScheme is a sharding scheme
-type ShardScheme interface {
-	// Shard will return a shard for a given identifer
+// ShardSet contains a sharding function and a set of shards, this interface allows for potentially out of order shard sets
+type ShardSet interface {
+	// Shard will return a shard for a given identifier
 	Shard(identifer string) uint32
 
 	// CreateSet will return a new shard set from a set of shards
 	CreateSet(shards []uint32) (ShardSet, error)
```

Comment:
How do you create a subset from a top level shardset then with the same `HashFn`?  Perhaps at least add a method "HashFn() HashFn" which returns the `HashFn` so that someone can then create a subset of the shards with the same `HashFn`.